client_test/rpc_test.go

```diff
@@ -87,18 +82,27 @@ func TestRpcClient(t *testing.T) {
 		testdata.Outcomes.Create(alice.Address(), bob.Address(), 100, 100, types.Address{}))
 
 	assert.Regexp(t, "VirtualFund.0x.*", vRes.Id)
-	waitTimeForCompletedObjectiveIds(t, &clientA, defaultTimeout, vRes.Id)
+
+	waitForObjectiveCompletion(rpcClientA, vRes.Id)
 	waitTimeForCompletedObjectiveIds(t, &clientB, defaultTimeout, vRes.Id)
 	waitTimeForCompletedObjectiveIds(t, &clientI, defaultTimeout, vRes.Id)
 	rpcClientA.Pay(vRes.ChannelId, 1)
 
 	closeVId := rpcClientA.CloseVirtual(vRes.ChannelId)
-	waitTimeForCompletedObjectiveIds(t, &clientA, defaultTimeout, closeVId)
+	waitForObjectiveCompletion(rpcClientA, closeVId)
 	waitTimeForCompletedObjectiveIds(t, &clientB, defaultTimeout, closeVId)
 	waitTimeForCompletedObjectiveIds(t, &clientI, defaultTimeout, closeVId)
 
 	closeId := rpcClientA.CloseLedger(res.ChannelId)
-	waitTimeForCompletedObjectiveIds(t, &clientA, defaultTimeout, closeId)
+	waitForObjectiveCompletion(rpcClientA, closeId)
 	waitTimeForCompletedObjectiveIds(t, &clientI, defaultTimeout, closeId)
 
 }
+
+func waitForObjectiveCompletion(c *rpc.RpcClient, expectedObjectiveId protocols.ObjectiveId) {
```

Comment:
This could be a method on `RpcClient`?

Reply:
[12a4f5c](https://github.com/statechannels/go-nitro/pull/1120/commits/12a4f5c582dbec1aff9e62bc1c642ed2b4bca989)